pkg/hwinfo/hwinfo_other.go

```diff
@@ -0,0 +1,197 @@
+// +build !darwin,!windows
+
+package hwinfo
+
+import (
+	"bufio"
+	"bytes"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"regexp"
+	"strconv"
+	"strings"
+
+	"github.com/cloudradar-monitoring/cagent/pkg/common"
+	"github.com/jaypipes/ghw"
+	"github.com/vcraescu/go-xrandr"
+)
+
+var lsusbLineRegexp = regexp.MustCompile(`[0-9|a-z|A-Z|.|/|-|:|\[|\]|_|+| ]+`)
+
+func captureStderr(funcToExecute func()) (string, error) {
+	r, w, err := os.Pipe()
+	if err != nil {
+		return "", err
+	}
+
+	stderr := os.Stderr
+	os.Stderr = w
+	defer func() {
+		os.Stderr = stderr
+	}()
+
+	funcToExecute()
+	err = w.Close()
+	if err != nil {
+		return "", err
+	}
+
+	var buf bytes.Buffer
+	_, err = io.Copy(&buf, r)
+	if err != nil {
+		return "", err
+	}
+
+	return buf.String(), nil
+}
+
+func listPCIDevices(errs *common.ErrorCollector) []*pciDeviceInfo {
+	var ghwErr error
+	var devices []*ghw.PCIDevice
+
+	// unfortunately, jaypipes/ghw library sometime writes error message directly to os.Stderr instead of returning from function call.
+	// that's why we will try to capture stderr output and handle it:
```

Comment:
Maybe vendor it with some fix inside? The stderr capturing looks too hacky.

Reply:
> The stderr capturing looks too hacky

I agree but vendoring it with a fix, forces us to "maintain" that fork, right?

Reply:
It depends really, I don't know how frequently this particular package is updated. And we'll be vendoring it anyway. Usually task can be simplified by creating a patch, so it could be applied when you bump vendored version of the target package. I still believe it's more worthy than hijacking Stderr.

Reply:
> I don't know how frequently this particular package is being updated. 

Have a look at the recent commits, there is some recent activity.
Having a patch and apply it over and over again introduces a new workflow which we don't have at the moment. So I'm not a huge fan of this.
Having this hack, at least keep it entirely in our codebase and we still get updates from upstream the normal way.